### PR TITLE
Upgrade to Jest 23

### PR DESCRIPTION
**Pull request by SimenB**
👋

I upgraded in order to check for regressions. And as I made the migration, I thought I might as well push it and make a PR.

This also includes some refatoring moving from jasmine based spies to the Jest ones.

I started migrating your `spyOnDev` stuff, but that was... a lot. So I pulled out. Sorry!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -69,8 +69,8 @@
     "gzip-js": "~0.3.2",
     "gzip-size": "^3.0.0",
     "jasmine-check": "^1.0.0-rc.0",
-    "jest": "^22.4.4",
-    "jest-diff": "^22.4.3",
+    "jest": "^23.0.0-charlie.4",
```
Comment (thymikee) [inline package.json]: Let's change to `23.0.1`

**Comment by gaearon** [comment]
Sorry we missed this, do you mind fixing up merge conflicts?

**Comment by SimenB** [comment]
@gaearon rebased and upgraded to 23.0.1

**Comment by SimenB** [comment]
`spyOnDev` and `spyOnProd` uses jasmine spies. Maybe @aaronabramov can codemod it?

**Comment by gaearon** [comment]
Thanks!

**Pull request merged**